### ISSUE TITLE
Support for Oracle procedure call from a package referenced via a database link

### ISSUE THREAD

**Issue by jpuris** [sqlfluff/sqlfluff#5043]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

When calling a function via a database link, SQLFluff seems to not understand what it is and assumes it to be a table reference.

### Use case

We use oracle database link to transfer data from `Oracle A` to `Oracle B`. dbt is used to run the transfer queries and we lint the models with SQLint.

In one of the relations, we have encrypted data, which can be decrypted with a procedure available in a package on source.

dbt has no issue running the model, however SQLfluff is not able to lint it without throwing various false positive errors and unparsables.

```
❯ sqlfluff lint dbt_project/models/stg/stg__customers_dblink.sql
=== [dbt templater] Sorting Nodes...
06:40:54  oracle adapter: Running in thin mode
06:40:54  Registered adapter: oracle=1.5.3
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [dbt_project/models/stg/stg__customers_dblink.sql] FAIL
L:   2 | P:   5 | RF03 | Unqualified reference 'id' found in single table select.
                       | [references.consistent]
L:   3 | P:  56 | RF01 | Reference 'my_schema.pkg_enc.decrypt_date'
                       | refers to table/view not found in the FROM clause or
                       | found in ancestor statement.
                       | [references.from]
L:   3 | P:  76 |  PRS | Line 3, Position 43: Found unparsable section:
                       | '@my_dblink(\n        birth_date_enc...'
All Finished 📜 🎉!
```

_False violations detected_

1. `RF03` this is detected because of `RF01`
2. `RF01` is due to it not understanding a package function call, where it assumes it is qualified table reference
3. `PRS` it does not understand the package function call with a database link

the dbt model is

```
SELECT
    id AS customer_id,
    {{ source('my_source', 'my_table').schema }}.pkg_enc.decrypt_date@my_dblink(
        birth_date_enc,
        birth_date_iv
    ) AS birth_date
FROM {{ source('my_source', 'my_table') }}
```

and compiled version looks as such

```
SELECT
    id AS customer_id,
    my_schema.pkg_enc.decrypt_date@my_dblink(
        birth_date_enc,
        birth_date_iv
    ) AS birth_date
FROM my_schema.my_table@my_dblink
```

### Dialect

Oracle

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
